servicetalk-http-api/src/main/java/io/servicetalk/http/api/HttpSerializers.java

```diff
@@ -0,0 +1,375 @@
+     * Creates an {@link HttpSerializerDeserializer} that uses {@link SerializerDeserializer} for serialization.
+     *
+     * @param serializer Used to serialize each {@link T}.
+     * @param headersSerializeConsumer Sets the headers to indicate the appropriate encoding and content type.
+     * @param headersDeserializePredicate Validates the headers are of the supported encoding and content type.
+     * @param <T> Type of object to serialize.
+     * @return {@link HttpSerializerDeserializer} that uses a {@link SerializerDeserializer} for serialization.
+     */
+    public static <T> HttpSerializerDeserializer<T> serializer(
+            SerializerDeserializer<T> serializer, Consumer<HttpHeaders> headersSerializeConsumer,
+            Predicate<HttpHeaders> headersDeserializePredicate) {
+        return new DefaultHttpSerializerDeserializer<>(serializer, headersSerializeConsumer,
+                headersDeserializePredicate);
+    }
+
+    /**
+     * Creates an {@link HttpStreamingSerializerDeserializer} that uses {@link StreamingSerializerDeserializer} for
+     * serialization.
+     *
+     * @param serializer Used to serialize each {@link T}.
+     * @param headersSerializeConsumer Sets the headers to indicate the appropriate encoding and content type.
+     * @param headersDeserializePredicate Validates the headers are of the supported encoding and content type.
+     * @param <T> Type of object to serialize.
+     * @return {@link HttpStreamingSerializerDeserializer} that uses a {@link StreamingSerializerDeserializer} for
+     * serialization.
+     */
+    public static <T> HttpStreamingSerializerDeserializer<T> streamingSerializer(
+            StreamingSerializerDeserializer<T> serializer, Consumer<HttpHeaders> headersSerializeConsumer,
+            Predicate<HttpHeaders> headersDeserializePredicate) {
+        return new DefaultHttpStreamingSerializerDeserializer<>(serializer, headersSerializeConsumer,
+                headersDeserializePredicate);
+    }
+
+    /**
+     * Create a {@link HttpStreamingSerializer} that uses a {@link Serializer} for serialization. This method is useful
+     * if the payload body is provided in non-{@link Buffer} type and the {@link HttpHeaderNames#CONTENT_TYPE} is known
+     * a-prior (e.g. streaming raw json data from a stream of {@link String}s). Deserialization should be done using
+     * the a-prior knowledge to use a compatible {@link HttpStreamingDeserializer}.
+     * @param serializer Used to serialize each {@link T} chunk.
+     * @param bytesEstimator Provides an estimate of how many bytes to allocate for each {@link Buffer} to serialize to.
+     * @param headersSerializeConsumer Sets the headers to indicate the appropriate encoding and content type.
+     * @param <T> Type of object to serialize.
+     * @return a {@link HttpStreamingSerializer} that uses a {@link Serializer} for serialization.
+     */
+    public static <T> HttpStreamingSerializer<T> streamingSerializer(
```

Comment:
There is a method above that returns `HttpStreamingSerializerDeserializer`, this method returns `HttpStreamingSerializer`. Do we need a method that returns `HttpStreamingDeserializer`?

Reply:
This was intended to support the same use case as described in https://github.com/apple/servicetalk/pull/1673#discussion_r684544839 ... if folks need deserialization specific variants we can always add more methods later, or they can use the `HttpStreamingSerializerDeserializer` variants and no-op the serialization side as short term approach.